lib/Parse/ParseExpr.cpp

```diff
@@ -661,7 +661,7 @@ ParserResult<Expr> Parser::parseExprSelector() {
     if (Tok.is(tok::r_paren))
       rParenLoc = consumeToken();
     else
-      rParenLoc = Tok.getLoc();
+      rParenLoc = PreviousLoc;
```

Comment:
This is a touch controversial.  In each case, the Verifier was smacking up against `Tok.getLoc()` here also being the starting location of the next declaration that got parsed since it's using a strict "x comes before y" check instead of `<=`.  But considering nothing important uses this SourceLoc I think it should fly.